src/Service/Security/SecurityChecker/SensioLabsSecurityChecker.php

```diff
@@ -5,20 +5,210 @@
+
+    /**
+     * @return Package[]
+     */
+    private function getPackages(string $lockFile): array
     {
-        // TODO: implement
+        $contents = json_decode($lockFile, true);
+        if (json_last_error() !== JSON_ERROR_NONE) {
+            throw new \UnexpectedValueException('Invalid composer.lock');
+        }
+
+        $packages = [];
+        foreach (['packages', 'packages-dev'] as $key) {
+            if (!isset($contents[$key]) || !is_array($contents[$key])) {
+                continue;
+            }
+
+            foreach ($contents[$key] as $package) {
+                $packages[] = new Package($package['name'], $package['version']);
+            }
+        }
+
+        return $packages;
+    }
+
+    /**
+     * @return array<string,Advisory[]>
+     */
+    private function getAdvisories(): array
+    {
+        $advisories = [];
+        foreach ($this->getDatabase() as $file) {
+            if (!$file->isFile() || $file->getExtension() !== 'yaml') {
+                continue;
+            }
+
+            $packageName = $this->parsePackageName(
+                str_replace($this->databaseDir, '', $file->getPathname())
+            );
+
+            if ($packageName === null) {
+                continue;
+            }
+
+            $data = $this->yamlParser->parse(
+                (string) file_get_contents($file->getRealPath())
+            );
+
+            if (!isset($advisories[$packageName])) {
+                $advisories[$packageName] = [];
+            }
+
+            $advisories[$packageName][] = new Advisory(
+                $data['title'],
+                $data['cve'] ?? '',
+                $data['link'],
+                array_map(
+                    fn ($branch) => new Versions(...$branch['versions']),
+                    array_values($data['branches'])
+                )
+            );
+        }
+
+        return $advisories;
+    }
+
+    private function getDatabase(): \RecursiveIteratorIterator
+    {
+        if (!is_dir($this->databaseDir)) {
+            throw new \RuntimeException('Advisories database does not exist');
+        }
+
+        $advisoryFilter = function (\SplFileInfo $file): bool {
+            if ($file->isDir()) {
+                $dirName = $file->getFilename();
+                if ($dirName[0] == '.') {
+                    return false;
+                }
+            }
+
+            return true;
+        };
+
+        return new \RecursiveIteratorIterator(
+            new \RecursiveCallbackFilterIterator(
+                new \RecursiveDirectoryIterator($this->databaseDir),
+                $advisoryFilter
+            )
+        );
+    }
+
+    private function parsePackageName(string $path): ?string
+    {
+        $matches = [];
+        preg_match('~^/(?<name>.+/.+)/.+yaml$~', $path, $matches);
+
+        return $matches['name'] ?? null;
+    }
+
+    private function loadAdvisoriesDatabase(): void
+    {
+        $this->advisories = $this->getAdvisories();
```

Comment:
maybe we should check if is already loaded? (to prevent load twice in one request)